image/image.go

```diff
@@ -109,14 +109,18 @@ func (i *genericImage) DockerTar() ([]byte, error) {
 // will support v1 one day...
 type genericManifest interface {
 	String() string
-	GetLayers() []string
+	Layers() []string
+	Config() ([]byte, error)
 	ImageInspectInfo() (*types.ImageInspectInfo, error)
 }
 
 type fsLayersSchema1 struct {
 	BlobSum string `json:"blobSum"`
 }
 
+// compile-time check that manifestSchema1 implements genericManifest
+var _ genericManifest = (*manifestSchema1)(nil)
```

Comment:
These checks should be unnecessary, the return type of `getParsedManifest` already allows the compiler to check this. But it also doesn’t hurt.